src/cpp/fastdds/xtypes/dynamic_types/DynamicDataImpl.cpp

```diff
@@ -0,0 +1,7218 @@
+                                    selected_union_member_ = MEMBER_ID_INVALID;
+                                    ReturnCode_t dyn_ret_value = get_int64_value(discriminator, 0);
+                                    if (RETCODE_OK != dyn_ret_value)
+                                    {
+                                        dyn_ret_value = get_uint64_value(udiscriminator, 0);
+                                        discriminator = static_cast<int64_t>(udiscriminator);
+                                    }
+
+                                    if (RETCODE_OK == dyn_ret_value)
+                                    {
+                                        for (auto member : type->get_all_members_by_index())
+                                        {
+                                            auto m_impl {traits<DynamicTypeMember>::narrow<DynamicTypeMemberImpl>(
+                                                             member)};
+
+                                            for (auto label : m_impl->get_descriptor().label())
+                                            {
+                                                if (static_cast<int32_t>(discriminator) == label)
+                                                {
+                                                    selected_union_member_ = m_impl->get_id();
+                                                    break;
+                                                }
+                                            }
+                                        }
+
+                                        if (MEMBER_ID_INVALID == selected_union_member_)
+                                        {
+                                            selected_union_member_ = type->default_union_member();
+                                        }
+                                    }
+                                    else
+                                    {
+                                        throw fastcdr::exception::BadParamException("Wrong discriminator");
+                                    }
+                                }
+                                break;
+                            default:
+                                {
+                                    if (1 == value_.count(selected_union_member_))
+                                    {
+                                        // Check MemberId in mutable case.
+                                        auto member_data {std::static_pointer_cast<DynamicDataImpl>(value_.at(
+                                                              selected_union_member_))};
+                                        dcdr >> member_data;
+                                        if (ExtensibilityKind::MUTABLE != type->get_descriptor().extensibility_kind())
+                                        {
+                                            ret_value = false;
+                                        }
```

Comment:
It would be nice to have some documentation here. Also explaining that the return value is not an error code but a flag to inform the user if the deserialization has to keep going.

Reply:
Done in 59043af58